httpbp/middlewares.go

```diff
@@ -158,3 +160,40 @@ func InjectEdgeRequestContext(truster HeaderTrustHandler, impl *edgecontext.Impl
 		}
 	}
 }
+
+// SupportedMethods returns a middleware that checks if the request is made
+// using one of the given HTTP methods.
+//
+// Returns a raw, plain text 405 error response if the method is not supported.
+// If GET is supported, HEAD will be automatically supported as well.
+func SupportedMethods(method string, additional ...string) Middleware {
+	supported := make(map[string]bool, len(additional)+1)
+	supported[strings.ToUpper(method)] = true
+	for _, m := range additional {
+		supported[strings.ToUpper(m)] = true
+	}
+
+	hasGet := false
+	for m := range supported {
+		hasGet = hasGet || strings.Compare(m, http.MethodGet) == 0
+	}
```

Comment:
you don't need this for loop. you already built the map, so you can just do a map lookup :)

but implicitly adding HEAD means that if some service really want to allow GET and disallow HEAD, there's no way they can achieve that with this middleware. I don't really know if that's a legit case in real life, though.

Reply:
🤦 

As far as not allowing HEAD... I'm not sure there's _really_ a use case for that and if there is, that endpoint can just not use this middleware and make their own.